engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/notempty/NotEmptyValidatorForArray.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.bv.notempty;
+
+import java.lang.reflect.Array;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraints.NotEmpty;
+
+/**
+ * Check that the array is not null and not empty.
+ *
+ * @author Guillaume Smet
+ */
+public class NotEmptyValidatorForArray implements ConstraintValidator<NotEmpty, Object[]> {
+
+	/**
+	 * Checks the array is not {@code null} and not empty.
+	 *
+	 * @param array the array to validate
+	 * @param constraintValidatorContext context in which the constraint is evaluated
+	 * @return returns {@code true} if the array is not {@code null} and the array is not empty
+	 */
+	@Override
+	public boolean isValid(Object[] array, ConstraintValidatorContext constraintValidatorContext) {
+		if ( array == null ) {
+			return false;
+		}
+		return Array.getLength( array ) > 0;
```

Comment:
Consistency, it's based on the Size validators. But I agree we should use array.length instead. Pushed an additional commit to fix it for Size and NotEmpty.